### PR TITLE
Add tests and fix arguments for manual report

### PR DESCRIPTION
**Pull request by ofrobots**
Positional optional arguments required a bit rigidity.

### CODE REVIEW

File: tests/unit/testManualHandler.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+test('undefined additional message', function(t) {
+  var r = report('Mickey', { method: 'SNIFF'}, undefined, function(err, res) {
+    t.end();
+  });
+  t.equal(r.message, 'Mickey', 'string error should propagate');
+  t.equal(r.context.httpRequest.method, 'SNIFF');
+});
+
+test('arguments after the callback funciton should get ignored', function(t) {
+  var r = report('honkey', function(err, res) {
```
Comment (cristiancavalli) [inline tests/unit/testManualHandler.js]: @ofrobots Maybe a different term? This is a bit controversial

**Comment by cristiancavalli** [comment]
LGTM - but could the test cases include the if that coveralls found? Seems like an good branch to test

File: tests/unit/testManualHandler.js (hunk 1 above)
Comment (ofrobots) [inline tests/unit/testManualHandler.js]: Thanks for catching this! I have changed it. 😨

**Comment by cristiancavalli** [comment]
Looks great!

**Comment by ofrobots** [comment]
Fixed a test to properly document check a specific edge case behaviour. Please take another look at the latest commit.

**Comment by cristiancavalli** [comment]
@ofrobots thanks for edge-case documentation - this LGTM
